docs/adr/decisions/0013-timeline-exposed-by-flows.md

```diff
@@ -0,0 +1,66 @@
+---
+status: "proposed"
+---
+# Timeline exposed by Flows
+
+## Context and Problem Statement
+
+The TAMS API specification and associated README do not explicitly indicate how the timings relate to timestamps inside media object files: the [README](https://github.com/bbc/tams/blob/16ea401/README.md#flow-and-media-timelines) simply calls it the "media timeline".
+A number of common codecs support temporal re-ordering which creates multiple timelines, for example the decode and presentation order of frames of video.
+To avoid confusion between clients and when moving between codecs, the TAMS API should define the relationship between these timelines.
+
+## Decision Drivers
+
+* All Flows of a Source share a common timeline, regardless of how they are coded, and clients need a common understanding to avoid "jumps" when changing codec
+* TAMS should aim to be flexible, while aligning with how other media formats behave (_e.g._ taking inspiration from MPEG-DASH and HLS)
+
+## Considered Options
+
+* Option 1: Make the Flow timeline represent the decode timeline
+* Option 2: Make the Flow timeline represent the presentation timeline
+* Option 2a: Additionally recommend Flow Segments contain whole GOPs or equivalent
+
+## Decision Outcome
+
+Chosen option: Option 2a: Make the Flow timeline represent the presentation timeline, and contain whole GOPs or equivalent, because it avoids problems where timing "jumps" when changing codec, without unduly complicating client implementations.
+
+### Implementation
+
+Implemented by <https://github.com/bbc/tams/pulls/37>
+
+## Pros and Cons of the Options
+
+### Option 1: Make the Flow timeline represent the _decode_ timeline
+
+Explicitly document that the Flow timeline represents the _decode_ timeline.
+Clients are expected to start reading either from the position of interest on the Flow timeline, or the first segment preceding that where `key_frame_count` is greater than 0.
+In either case, a client may need to read an earlier segment in order to locate the timestamp of interest due to temporal re-ordering.
+This is how the BBC R&D proof-of-concept implementation works at time of writing, largely for historical reasons.
+
+* Good, because regardless of Flow Segment size, the ["media timeline"](https://github.com/bbc/tams/blob/16ea401/README.md#flow-and-media-timelines) inside the segment contains monotonically increasing timestamps.
+* Good, because it aligns with how the existing proof-of-concept implementation works.
+* Bad, because it's the opposite of how chunked streaming formats like MPEG-DASH and HLS work, which refer to the presentation timeline.
+* Bad, because transcoding may introduce or adjust decoding delay, so for example multiple Flows of the same Source may have frames with the same decode timestamp, but different presentation timestamps.
```

Comment:
Going slightly further, a (decoding) timestamp for a Source frame may end up referencing different frames in Flows (e.g. a intra-coded Flow vs a temporally re-ordered inter-coded Flow). This breaks precise time addressability (and immutability?).

Reply:
That's exactly what I'm trying to say, but I've had another go (borrowing quite a few of your words to do so 😄) in b356793